src/common/simple_spin.cc

```diff
@@ -18,9 +18,12 @@
 #include <stdint.h>
 #include <pthread.h>
 
-static uint32_t bar = 13;
-static uint32_t *foo = &bar;
- 
+#if (__arm__ || __aarch64__)
+#define cpu_relax() asm volatile("yield": : :"memory")
```

Comment:
Nearly all the examples I found online include the memory barrier but I'll remove it as per your suggestion.